doc/arkode/guide/source/Usage/SPRKStep_c_interface/User_callable.rst

```diff
@@ -0,0 +1,1309 @@
+   :param t0: the initial value of :math:`t`.
+   :param y0: the initial condition vector :math:`y(t_0)`.
+
+   :return:
+      * *ARK_SUCCESS* if successful
+      * *ARK_MEM_NULL*  if the SPRKStep memory was ``NULL``
+      * *ARK_MEM_FAIL*  if a memory allocation failed
+      * *ARK_ILL_INPUT* if an argument has an illegal value.
+
+   **Notes:**
+      All previously set options are retained but may be updated by calling
+      the appropriate "Set" functions.
+
+      If an error occurred, :c:func:`SPRKStepReInit()` also
+      sends an error message to the error handler function.
+
+
+
+
+.. _ARKODE.Usage.SPRKStep.Reset:
+
+SPRKStep reset function
+-----------------------
+
+To reset the SPRKStep module to a particular state :math:`(t_R,y(t_R))` for the
+continued solution of a problem, where a prior
+call to :c:func:`SPRKStepCreate` has been made, the user must call the function
+:c:func:`SPRKStepReset()`.  Like :c:func:`SPRKStepReInit()` this routine retains
+the current settings for all SPRKStep module options and performs no memory
+allocations but, unlike :c:func:`SPRKStepReInit()`, this routine performs only a
+*subset* of the input checking and initializations that are done in
+:c:func:`SPRKStepCreate`. In particular this routine retains all internal
+counter values. Like :c:func:`SPRKStepReInit()`, a call to
+:c:func:`SPRKStepReset()` will delete any previously-set *tstop* value specified
+via a call to :c:func:`SPRKStepSetStopTime()`.  Following a successful call to
+:c:func:`SPRKStepReset()`, call :c:func:`SPRKStepEvolve()` again to continue
+solving the problem. By default the next call to :c:func:`SPRKStepEvolve()` will
+use the step size computed by SPRKStep prior to calling :c:func:`SPRKStepReset()`.
+To set a different step size or have SPRKStep estimate a new step size use
+:c:func:`SPRKStepSetInitStep()`.
+
+One important use of the :c:func:`SPRKStepReset()` function is in the
+treating of jump discontinuities in the RHS functions.  Except in cases
+of fairly small jumps, it is usually more efficient to stop at each
+point of discontinuity and restart the integrator with a readjusted
+ODE model, using a call to :c:func:`SPRKStepReset()`.  To stop when
+the location of the discontinuity is known, simply make that location
+a value of ``tout``.  To stop when the location of the discontinuity
+is determined by the solution, use the rootfinding feature.  In either
+case, it is critical that the RHS functions *not* incorporate the
+discontinuity, but rather have a smooth extension over the
+discontinuity, so that the step across it (and subsequent rootfinding,
+if used) can be done efficiently.  Then use a switch within the RHS
+functions (communicated through ``user_data``) that can be flipped
+between the stopping of the integration and the restart, so that the
+restarted problem uses the new values (which have jumped).  Similar
+comments apply if there is to be a jump in the dependent variable
+vector.
```

Comment:
I recommend that we just remove this paragraph, since it matches what was said above for ReInit, and doesn't really apply for just resetting an initial condition.